tests/java-tests/src/test/java/test/ClientTest.java

```diff
@@ -41,8 +67,153 @@ public void before() {
   @Test
   public void simple() {
     mockServer.enqueue(new MockResponse("{\"data\": {\"random\": 42}}"));
+    ApolloResponse<GetRandomQuery.Data> queryResponse = Rx2Apollo.rxSingle(
+        apolloClient.query(new GetRandomQuery())
+    ).blockingGet();
+    Truth.assertThat(queryResponse.dataAssertNoErrors().random).isEqualTo(42);
+
+    mockServer.enqueue(new MockResponse("{\"data\": {\"createAnimal\": {\"__typename\": \"Cat\", \"species\": \"cat\", \"habitat\": {\"temperature\": 10.5}}}}"));
+    ApolloResponse<CreateCatMutation.Data> mutationResponse = Rx2Apollo.rxSingle(
+        apolloClient.mutate(new CreateCatMutation())
+    ).blockingGet();
+    Truth.assertThat(mutationResponse.dataAssertNoErrors().createAnimal.catFragment.species).isEqualTo("cat");
+
+    Disposable disposable = Rx2Apollo.rxFlowable(
+        apolloClient.subscribe(new AnimalCreatedSubscription())
+    ).subscribe(result -> {
+      String species = result.dataAssertNoErrors().animalCreated.catFragment.species;
+    });
+  }
+
+  private void autoPersistedQueries() {
+    apolloClient = new ApolloClient.Builder()
+        .serverUrl("https://localhost")
+        .autoPersistedQueries()
+        .build();
+  }
+
+  private void queryBatching() {
+    apolloClient = BatchingHttpEngineExtensions.canBeBatched(
+        new ApolloClient.Builder()
+            .serverUrl("https://localhost")
+            .httpEngine(new BatchingHttpEngine()),
+        false
+    ).build();
+
+    ApolloResponse<GetRandomQuery.Data> result = Rx2Apollo.rxSingle(BatchingHttpEngineExtensions.canBeBatched(
+        apolloClient.query(new GetRandomQuery()),
+        true
+    )).blockingGet();
+  }
+
+  private void httpCache() {
+    File cacheDir = new File("/tmp/apollo-cache");
+    long cacheSize = 10_000_000;
+    apolloClient = HttpCacheExtensions.httpCache(
+        new ApolloClient.Builder().serverUrl("https://localhost"),
+        cacheDir,
+        cacheSize
+    ).build();
```

Comment:
What about something like this instead?

```java
ApolloClient.Builder builder = new ApolloClient.Builder().serverUrl("https://localhost")
ApolloClientBuilder.httpCache(builder, cacheDir, cacheSize)
ApolloClient apolloClient = builder.build()
```

Reply:
Done on b20a3581b2ab8067f715741c8a595c3b6f79818c. What do you think?